### PR TITLE
Moving fuzzy-native to core

### PR DESCRIPTION
**Pull request by mauricioszabo**
Creating a `atom.fuzzyMatcher` API containing `setCandidates(arrayOfCandidates)`, `score(candidate, stringToMatch)` and `match(candidate, stringToMatch)`.

Also fixed fuzzy-finder, command-palette, and autocomplete-plus to use this API instead of adding hundreds of different fuzzy-finder dependencies, each on its own version

### CODE REVIEW

**Comment by asiloisad** [comment]
I'm super glad this PR. It has to be done long time ago as many packages want to use 👍

**Review by confused-Techie — APPROVED** [review]
Seriously this looks fantastic!

I love the way you've exported the `fuzzyMatcher` object with all of it's functions as a part of that one object. Really helps to compartmentalize the whole thing, and I'll absolutely be taking the same queue for the markdown portion.

Otherwise, this is amazing to see such a simple drop in replacement for the functionality in so many packages, (plus glad you updated the relevant settings on each one) so it's awesome to see.

I'd be happy getting this one merged, although of course I still have some work to do over on my portion of the branch, maybe for ease of reviewing my markdown changes we wait until I'm done over there prior to merging?

**Comment by mauricioszabo** [comment]
Sure, @confused-Techie, let's wait for your portion and then we can merge it :)

**Review by DeeDeeG — COMMENTED** [review]
I'm "lite-approve"-ing this, since it has popular demand behind it.

Once again, it seems a bit above my ability to review every bit of the diff, but I offer to help with troubleshooting if there is any issue with it post-merge.

**Comment by confused-Techie** [comment]
@mauricioszabo If you want to go ahead and take this one out of draft status, feel free my other PR has been approved.

Also the failing CI here seems to be from a mistake on my end, that's now been resolved.

Also just now realized with this PR targeting my branch specifically may be problematic, sorry about that. Hopefully you can just target the main branch now, but not sure what's easiest on your end

**Comment by mauricioszabo** [comment]
@confused-Techie great, it already updated to `master` so it's all good :)

I'll just add some JSDocs too that I forgot

File: src/ui.js [hunk 1]
```diff
@@ -2,6 +2,8 @@ const path = require("path");
 const fs = require("fs");
 const MarkdownIt = require("markdown-it");
 const { TextEditor } = require("atom");
+const { Matcher } = require("@pulsar-edit/fuzzy-native");
+let yamlFrontMatter, markdownItEmoji, markdownItGitHubHeadings, markdownItTaskCheckbox;
```
Comment (confused-Techie) [inline src/ui.js:6]:
```suggestion
```

Seems a merge somewhere went a little astray. This string of variables is no longer needed by the `render()` function

**Review by confused-Techie — APPROVED** [review]
This looks good to me, approval part 2

**Pull request merged**